### PR TITLE
Removed broken links

### PR DESCRIPTION
**Pull request by julian-st**
## What does this PR do?
Removed broken links for ruby on rails and visual basic 2010 book, alternatives are already there

## Checklist:
- [x ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ x] Include author(s) and platform where appropriate.
- [ x] Put lists in alphabetical order, correct spacing.
- [ x] Add needed indications (PDF, access notes, under construction).
- [x ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-de.md [hunk 1]
```diff
@@ -205,7 +205,6 @@
 
 ### Ruby on Rails
 
-* [Praxiswissen Ruby](http://www.oreilly.de/german/freebooks/rubybasger/pdf_rubybasger.pdf) (PDF)
```
Comment (LuigiImVector) [inline books/free-programming-books-de.md:208]: You can replace it with: https://web.archive.org/web/20160328183933/https://oreilly.de/german/freebooks/rubybasger/pdf_rubybasger.pdf

File: books/free-programming-books-de.md [hunk 2]
```diff
@@ -249,6 +248,5 @@
 
 ### Visual Basic
 
-* [Einstieg in Visual Basic 2010](http://openbook.rheinwerk-verlag.de/einstieg_vb_2010) - Thomas Theis (Online)
```
Comment (LuigiImVector) [inline books/free-programming-books-de.md:252]: You can replace it with: https://web.archive.org/web/20210507062754/https://openbook.rheinwerk-verlag.de/einstieg_vb_2010/

**Review by LuigiImVector — CHANGES_REQUESTED** [review]
Respect the [formatting guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md#formatting)

**Comment by LuigiImVector** [comment]
Related to #6942

File: books/free-programming-books-de.md (hunk 2 above)
Comment (davorpa) [reply books/free-programming-books-de.md:252]: How is it different from the 2012 version?

Comment (LuigiImVector) [reply books/free-programming-books-de.md]: 2012 version has more chapters, all right

**Pull request merged**